saleor/graphql/translations/mutations.py

```diff
@@ -0,0 +1,185 @@
+import graphene
+from django.core.exceptions import ImproperlyConfigured
+
+from ...discount import models as discount_models
+from ...menu import models as menu_models
+from ...page import models as page_models
+from ...product import models as product_models
+from ...shipping import models as shipping_models
+from ..core.mutations import (
+    BaseMutation, ModelMutationOptions, get_model_name, get_output_fields,
+    registry)
+
+
+class BaseTranslateMutation(BaseMutation):
+
+    @classmethod
+    def __init_subclass_with_meta__(
+            cls, arguments=None, model=None, _meta=None, **options):
+        if not model:
+            raise ImproperlyConfigured('No model provided in Meta')
+        if not _meta:
+            _meta = ModelMutationOptions(cls)
+        return_field_name = get_model_name(model)
+        fields = get_output_fields(model, return_field_name)
+
+        _meta.model = model
+        _meta.return_field_name = return_field_name
+        super().__init_subclass_with_meta__(_meta=_meta, **options)
+        cls._meta.fields.update(fields)
+
+    class Meta:
+        abstract = True
+
+    @classmethod
+    def mutate(cls, root, info, id, language_code, input):
+        errors = []
+        model_type = registry.get_type_for_model(cls._meta.model)
+        instance = cls.get_node_or_error(info, id, errors, 'id', model_type)
+
+        if errors:
+            return cls(errors=errors)
+
+        instance.translations.update_or_create(
+            language_code=language_code, defaults=input)
+        return cls(**{cls._meta.return_field_name: instance, 'errors': errors})
+
+
+class NameTranslateInput(graphene.InputObjectType):
```

Comment:
Sorry for nitpicking - don't you think `TranslationInput` would sound more correct for inputs? So we'd have `NameTranslationInput`, `SeoTranslationInput` etc. I think it sounds more natural:)

Reply:
For mutations, the naming pattern `<Object>Translate` sounds fine because we use a verb for action then.